lib/facter/custom_facts/core/execution/base.rb

```diff
@@ -35,10 +35,12 @@ def with_env(values)
           rv
         end
 
-        def execute(command, options = {})
+        def execute(command, options = {}) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
           on_fail = options.fetch(:on_fail, :raise)
           expand = options.fetch(:expand, true)
           logger = options[:logger]
+          time_limit = options[:limit].to_i
+          time_limit = time_limit.positive? ? time_limit : nil
```

Comment:
We can extract this block of code in another private method e.g.

```
         def extract_options(options)
          on_fail = options.fetch(:on_fail, :raise)
          expand = options.fetch(:expand, true)
          logger = options[:logger]
          time_limit = options[:limit].to_i
          time_limit = time_limit.positive? ? time_limit : nil

          [on_fail, expand, logger, time_limit]
        end
```

and use 
```
on_fail, expand, logger, time_limit = extract_options(options)
```

to get rid of the rubocop exceptions.

Reply:
Done